fast64_internal/f3d/f3d_parser.py

```diff
@@ -1361,7 +1364,9 @@ def loadTexture(self, data, name, region, tileSettings, isLUT):
 
         # TODO: Textures are sometimes loaded in with different dimensions than for rendering.
         # This means width is incorrect?
-        image = parseTextureData(data, textureName, self, tileSettings.fmt, siz, width, self.basePath, isLUT, self.f3d)
+        image, loadedFromImageFile = parseTextureData(data, textureName, self, tileSettings.fmt, siz, width, self.basePath, isLUT, self.f3d)
+        if loadedFromImageFile and image not in self.imagesLoadedFromFile:
+            self.imagesLoadedFromFile.append(image)
```

Comment:
I suggest making the collection of images a set (in constructor `self.imagesLoadedFromFile = set()`)
and renaming that collection to what it's used for, like `imagesDontApplyTlut` maybe?
```suggestion
        if loadedFromImageFile:
            self.imagesDontApplyTlut.add(image)
```
or reversing things may be simpler:
```py
        if not loadedFromImageFile:
            self.imagesApplyTlut.add(image)
```
(and `and (texProp.tex in self.imagesApplyTlut`)